docs/OffsetHelper.md

```diff
@@ -333,6 +333,562 @@ Retire the specified TCO2 tokens.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| [0] | uint256 | Resulting amount of Toucan pool token that got acquired for the swapped ERC20 tokens. |
+
+### fallback
+
+```solidity
+fallback() external payable
+```
+
+### receive
+
+```solidity
+receive() external payable
+```
+
+### calculateNeededETHAmount
+
+```solidity
+function calculateNeededETHAmount(address _toToken, uint256 _toAmount) public view returns (uint256)
+```
+
+Return how much MATIC is required in order to swap for the
+desired amount of a Toucan pool token, for example, BCT or NCT.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| _toToken | address | The address of the pool token to swap for, for example, NCT or BCT |
+| _toAmount | uint256 | The desired amount of pool token to receive |
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| [0] | uint256 | amounts The amount of MATIC required in order to swap for the specified amount of the pool token |
+
+### calculateExpectedPoolTokenForETH
+
+```solidity
+function calculateExpectedPoolTokenForETH(uint256 _fromMaticAmount, address _toToken) public view returns (uint256)
+```
+
+Calculates the expected amount of Toucan Pool token that can be
+acquired by swapping the provided amount of MATIC.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| _fromMaticAmount | uint256 | The amount of MATIC to swap |
+| _toToken | address | The address of the pool token to swap for, for example, NCT or BCT |
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| [0] | uint256 | The expected amount of Pool token that can be acquired |
+
+### swapExactOutETH
+
+```solidity
+function swapExactOutETH(address _toToken, uint256 _toAmount) public payable
+```
+
+Swap MATIC for Toucan pool tokens (BCT/NCT) on SushiSwap.
+Remaining MATIC that was not consumed by the swap is returned.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| _toToken | address | Token to swap for (will be held within contract) |
+| _toAmount | uint256 | Amount of NCT / BCT wanted |
+
+### swapExactInETH
+
+```solidity
+function swapExactInETH(address _toToken) public payable returns (uint256)
+```
+
+Swap MATIC for Toucan pool tokens (BCT/NCT) on SushiSwap. All
+provided MATIC will be swapped.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| _toToken | address | Token to swap for (will be held within contract) |
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| [0] | uint256 | Resulting amount of Toucan pool token that got acquired for the swapped MATIC. |
+
+### withdraw
+
+```solidity
+function withdraw(address _erc20Addr, uint256 _amount) public
+```
+
+Allow users to withdraw tokens they have deposited.
+
+### deposit
+
+```solidity
+function deposit(address _erc20Addr, uint256 _amount) public
+```
+
+Allow users to deposit BCT / NCT.
+
+_Needs to be approved_
```

Comment:
Does this refer to ERC20 `approve()`?  Would be better to be explicit about this.  If so, it would also be a nice touch to add support for [EIP-2612: Permit Extension for EIP-20 Signed Approvals](https://eips.ethereum.org/EIPS/eip-2612) so that an extra transaction is not required for `approve()`.  I don't think we can justify prioritizing this high on our backlog at the moment, but if you are looking for excuses to expand your Solidity skills it might be a great weekend hackathon challenge ;-)